mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -194,6 +220,29 @@ def _extract_headers_from_credentials(self, headers):
         """
         return {header.name: header.value for header in headers}
 
+    def _azure_upload_chunk(
+        self, credentials, headers, local_file, artifact_path, start_byte, size
+    ):
+        # Base64-encode a UUID, producing a UTF8-encoded bytestring. Then, decode
+        # the bytestring for compliance with Azure Blob Storage API requests
+        block_id = base64.b64encode(uuid.uuid4().hex.encode()).decode("utf-8")
+        chunk = read_chunk(local_file, size, start_byte)
```

Comment:
I found boto3 (s3transfer) uses a stream upload to avoid the entire chunk into memory. Not a blocker, but a nice enhancement (if it doesn't degrate the performance). The implementation cost is low, we can just replace `chunk` (bytes) with a file-like object.

Reply:
Makes sense, but probably not critical here initially. Thanks Haru!